src/pip/_internal/network/lazy_wheel.py

```diff
@@ -1,66 +1,122 @@
 """Lazy ZIP over HTTP"""
 
-__all__ = ["HTTPRangeRequestUnsupported", "dist_from_wheel_url"]
+from __future__ import annotations
 
+__all__ = ["HTTPRangeRequestUnsupported", "dist_from_wheel_url", "LazyWheelOverHTTP"]
+
+import abc
+import io
+import logging
+import re
 from bisect import bisect_left, bisect_right
 from contextlib import contextmanager
 from tempfile import NamedTemporaryFile
-from typing import Any, Dict, Generator, List, Optional, Tuple
+from typing import Any, BinaryIO, ClassVar, Iterable, Iterator, cast
+from urllib.parse import urlparse
 from zipfile import BadZipFile, ZipFile
 
 from pip._vendor.packaging.utils import canonicalize_name
-from pip._vendor.requests.models import CONTENT_CHUNK_SIZE, Response
+from pip._vendor.requests.models import CONTENT_CHUNK_SIZE, HTTPError, Response
+from pip._vendor.requests.status_codes import codes
 
+from pip._internal.exceptions import InvalidWheel, UnsupportedWheel
 from pip._internal.metadata import BaseDistribution, MemoryWheel, get_wheel_distribution
-from pip._internal.network.session import PipSession
-from pip._internal.network.utils import HEADERS, raise_for_status, response_chunks
+from pip._internal.network.session import PipSession as Session
```

Comment:
Why the name change?

Reply:
This code was re-used for a non-pip context.

Reply:
Fixed!